crates/wasmparser/src/validator/core/canonical.rs

```diff
@@ -118,6 +118,14 @@ pub(crate) trait InternRecGroup {
             }
         }
 
+        if is_new {
+            for i in start..end {
+                let i = u32::try_from(i).unwrap();
+                let id = CoreTypeId::from_index(i);
+                self.check_descriptors(rec_group_id, id, types, offset)?;
```

Comment:
Could this get folded into the `if is_new` block above? Or does this semantically need to happen after the above loop?